lib/compress/zstd_lazy.c

```diff
@@ -871,33 +871,40 @@ FORCE_INLINE_TEMPLATE size_t ZSTD_HcFindBestMatch_extDict_selectMLS (
 
 #define ZSTD_ROW_HASH_CACHE_MASK (ZSTD_ROW_HASH_CACHE_SIZE - 1)
 
-typedef U32 ZSTD_VecMask;   /* Clarifies when we are interacting with a U32 representing a mask of matches */
+typedef U64 ZSTD_VecMask;   /* Clarifies when we are interacting with a U64 representing a mask of matches */
```

Comment:
Something like a ~15% regression.

```
rebalanced 32-bit:
 5#silesia.tar       : 211957760 ->  62473463 (3.393),  72.5 MB/s , 596.1 MB/s 
 6#silesia.tar       : 211957760 ->  61461315 (3.449),  61.0 MB/s , 596.9 MB/s 
 7#silesia.tar       : 211957760 ->  60459438 (3.506),  47.6 MB/s , 629.6 MB/s 
 8#silesia.tar       : 211957760 ->  59989973 (3.533),  38.0 MB/s , 638.0 MB/s 
 9#silesia.tar       : 211957760 ->  59707605 (3.550),  35.9 MB/s , 644.7 MB/s 
10#silesia.tar       : 211957760 ->  59157938 (3.583),  33.5 MB/s , 644.0 MB/s 
11#silesia.tar       : 211957760 ->  58644580 (3.614),  25.3 MB/s , 652.9 MB/s 
12#silesia.tar       : 211957760 ->  58243087 (3.639),  18.6 MB/s , 660.0 MB/s

dev 32-bit with same params:
 5#silesia.tar       : 211957760 ->  62473463 (3.393),  78.8 MB/s , 593.5 MB/s 
 6#silesia.tar       : 211957760 ->  61461315 (3.449),  70.1 MB/s , 594.7 MB/s 
 7#silesia.tar       : 211957760 ->  60459438 (3.506),  52.9 MB/s , 627.3 MB/s 
 8#silesia.tar       : 211957760 ->  59989973 (3.533),  48.0 MB/s , 635.8 MB/s 
 9#silesia.tar       : 211957760 ->  59707605 (3.550),  42.4 MB/s , 642.4 MB/s 
10#silesia.tar       : 211957760 ->  59157938 (3.583),  39.6 MB/s , 641.8 MB/s 
11#silesia.tar       : 211957760 ->  58644580 (3.614),  31.8 MB/s , 650.4 MB/s 
12-silesia.tar       : 211957760 ->  58644580 (3.614),  31.6 MB/s , 650.6 MB/s
```

Reply:
I don't care about 32-bit builds...  but branching on `rowEntries` in `ZSTD_VecMask_next()` would probably recover most performance in the 16/32 bits case. In the 64-bits case, on 32-bit platforms, it would be faster to process each each 32-bit half separately (which doesn’t really fit with the current `ZSTD_VecMask_next()` abstraction).

Reply:
`matches &= (matches - 1)` could be pull into its own function ... that branches on `rowEntries` and `sizeof(size_t)`